sdk/core/Azure.Core/src/Shared/TaskExtensions.cs

```diff
@@ -87,5 +91,39 @@ private static void VerifyTaskCompleted(bool isCompleted)
                 throw new InvalidOperationException("Task is not completed");
             }
         }
+
+        public readonly struct Enumerable<T> : IEnumerable<T>
```

Comment:
Should these be moved to their own file?   They're not, strictly speaking, task extensions.

Reply:
It is part of the implementation. `List<T>` has its enumerable & enumerator in its own file: https://github.com/dotnet/corefx/blob/e99ec129cfd594d53f4390bf97d1d736cff6f860/src/Common/src/CoreLib/System/Collections/Generic/List.cs#L1077

Reply:
Should we at least make them `private` then?  It doesn't look like these should be used directly.

Reply:
They are used directly, but implicitly, by `foreach`. The benefit of `Enumerable<T>` and `Enumerator<T>` being structs is that `foreach` can use duck typing and call `Enumerator<T> GetEnumerator()` instead of the method declared in the `IEnumerable<T>` interface and avoid heap memory allocation. But that can only work if both `Enumerable<T>` and `Enumerator<T>` are available in the scope of that `foreach`.

Reply:
That makes sense.  Can we add a comment to that effect so nobody accidentally breaks this in the future?

Reply:
Fixed